synapse/cortex.py

```diff
@@ -1095,6 +1095,49 @@ async def _viewNomergeToProtected(self):
             await view.setViewInfo('protected', nomerge)
             await view.setViewInfo('nomerge', None)
 
+    async def _migrateCronUser(self):
+        for iden, cron in self.agenda.list():
+            await self.editCronJob(iden, 'user', cron.creator)
+
+        oldperm = ('cron', 'set', 'creator')
+        newperm = ('cron', 'set', 'user')
+
+        for user in self.auth.users():
+            rules = []
+            update = False
+
+            for allow, path in user.getRules():
+                if path[:3] == oldperm:
+                    update = True
+                    rules.append((allow, newperm + path[3:]))
+                    continue
+                rules.append((allow, path))
+
+            if update:
+                await user.setRules(rules)
+
+        for role in self.auth.roles():
+            rules = []
+            update = False
+
+            for allow, path in role.getRules():
+                if path[:3] == oldperm:
+                    update = True
+                    rules.append((allow, newperm + path[3:]))
+                    continue
+                rules.append((allow, path))
+
+            if update:
+                await role.setRules(rules)
+
+    @s_nexus.Pusher.onPushAuto('cortex:add:trigger:creator')
```

Comment:
Is this the first time we've used a nexus event to trigger a storage migration? Should we consider a naming convention for them?

Reply:
We have used them in advanced power-up migrations (`'search:migr:v4'` for example`) but this is the first cortex one. Should probably decide on a naming convention, yeah.

Reply:
renamed to `cortex:migr:trigger:creator`